provider/testdata/src/lib.rs

```diff
@@ -102,42 +84,48 @@
 
 extern crate alloc;
 
-// If you want this to be made public stable, file an issue on the ICU4X repository.
-#[cfg(feature = "metadata")]
-#[doc(hidden)]
-pub mod metadata;
+#[path = "../metadata.rs.data"]
+mod metadata;
 
-#[cfg(feature = "metadata")]
 pub mod versions {
     //! Functions to access version info of the ICU test data.
 
     /// Gets the CLDR tag used as the test data source (for formatters, likely subtags, ...)
     ///
-    /// Enabled with the "metadata" Cargo feature.
-    ///
     /// # Examples
     ///
     /// ```
     /// assert_eq!("42.0.0", icu_testdata::versions::cldr_tag());
     /// ```
-    pub fn cldr_tag() -> String {
-        crate::metadata::load().cldr_json_gitref
+    pub const fn cldr_tag() -> &'static str {
+        crate::metadata::CLDR_JSON_GITREF
     }
 
     /// Gets the ICU tag used as the test data source (for properties, collator, ...)
     ///
-    /// Enabled with the "metadata" Cargo feature.
-    ///
     /// # Examples
     ///
     /// ```
     /// assert_eq!("release-72-1", icu_testdata::versions::icu_tag());
     /// ```
-    pub fn icu_tag() -> String {
-        crate::metadata::load().icuexportdata_gitref
+    pub const fn icu_tag() -> &'static str {
+        crate::metadata::ICUEXPORTDATA_GITREF
     }
 }
 
+/// Gets the locales supported by the test data.
+///
+/// # Examples
+///
+/// ```
+/// # use icu_locid::langid;
+/// assert!(icu_testdata::locales().contains(&langid!("es-AR")));
+/// assert!(icu_testdata::locales().len() > 10);
+/// ```
+pub const fn locales() -> &'static [icu_locid::LanguageIdentifier] {
```

Comment:
Issue: Even if we adopt a .data.rs file instead of .toml, I think I still prefer returning allocated vectors here because I don't want to lock us into a certain representation.